### PR TITLE
Don't enable address sanitizer in debug builds by default

### PR DESCRIPTION
**Pull request by Trial97**
Maybe it fixes stuff: 
image

### CODE REVIEW

**Review by TayouVR — APPROVED** [review]
I think this is good, we could consider enabling it in the future again after more rigorous testing. Making sure all CI actually builds, etc.

**Review by TheKodeToad — APPROVED** [review]
I think opt-in is more logical

**Review by txtsd — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -85,7 +85,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DTOML_ENABLE_FLOAT16=0")
 # set CXXFLAGS for build targets
 set(CMAKE_CXX_FLAGS_RELEASE "-O2 -D_FORTIFY_SOURCE=2 ${CMAKE_CXX_FLAGS_RELEASE}")
 
-option(DEBUG_ADDRESS_SANITIZER "Enable Address Sanitizer in Debug builds" on)
+option(DEBUG_ADDRESS_SANITIZER "Enable Address Sanitizer in Debug builds" off)
```
Comment (Scrumplex) [inline CMakeLists.txt]: nitpick:
```suggestion
option(DEBUG_ADDRESS_SANITIZER "Enable Address Sanitizer in Debug builds" OFF)
```

**Comment by Scrumplex** [comment]
Let's have this in staging instead, as it's causing issues there for mostly

**Pull request merged**